tests/validation/tests/v3_api/test_create_ha.py

```diff
@@ -295,13 +295,13 @@ def write_kubeconfig():
     file.close()
 
 
-def set_url_and_password(server_url):
-    admin_token = set_url_password_token(server_url)
-    admin_client = rancher.Client(url=server_url + "/v3",
+def set_url_and_password(rancher_url, server_url=None):
+    admin_token = set_url_password_token(rancher_url, server_url)
+    admin_client = rancher.Client(url=rancher_url + "/v3",
                                   token=admin_token, verify=False)
-    auth_url = server_url + "/v3-public/localproviders/local?action=login"
+    auth_url = rancher_url + "/v3-public/localproviders/local?action=login"
     user, user_token = create_user(admin_client, auth_url)
-    env_details = "env.CATTLE_TEST_URL='" + server_url + "'\n"
+    env_details = "env.CATTLE_TEST_URL='" + rancher_url + "'\n"
```

Comment:
in the case that `server_url` is NOT None, should we use `rancher_url` or `server_url`?

Reply:
`rancher_url` should be used in these places as that is still where rancher is accessible and `.../v3` uri lives.

Reply:
what is the difference between the `rancher_url` and the `server_url`?

Reply:
`rancher_url` is the url used to access rancher -- the public IP of the rancher server.

`server_url` is the default rancher install url. All of the nodes in a cluster must be able to reach that url and is used as the access point to rancher when adding clusters/nodes. 

Generally these are the same, but in the case of airgap testing we have them different so that we have an exposed rancher_url without doing SOCKS proxies, but clusters are still able to be added because they look at the other (private) URL